api/apps/geoprocessing/src/modules/protected-areas/dto/upload.shapefile.dto.ts

```diff
@@ -0,0 +1,9 @@
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsOptional, IsString } from 'class-validator';
+
+export class UploadShapefileDto {
+  @ApiPropertyOptional()
+  @IsString()
+  @IsOptional()
+  name?: string;
```

Comment:
This can be potentially dangerous:

https://nodejs.org/en/knowledge/file-system/security/introduction/

implementation in our codebase:
`api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts`

Reply:
sql injection ?

Reply:
may not apply if using parametrized queries